google/cloud/aiplatform/datasets/image_dataset.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Optional, Sequence, Dict, Tuple
+
+from google.auth import credentials as auth_credentials
+
+from google.cloud.aiplatform import schema
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform.datasets import (
+    Dataset,
+    NonTabularDatasource,
+    NonTabularDatasourceImportable,
+)
+
+
+class ImageDataset(Dataset):
+    """Managed image dataset resource for AI Platform"""
+
+    _support_metadata_schema_uris = (schema.dataset.metadata.image,)
+
+    _support_import_schema_classes = ("image",)
+
+    def __init__(
```

Comment:
Do we need an init method at all in the `ImageDataset` subclass? Users could just call `ImageDataset.create`.

Furthermore, it seems like this just calls the Dataset's init method with no modifications so wouldn't it be redundant?

Reply:
One reason for using an overridden init in the subclass is to check that the returned existing Dataset's `metadata_schema_uri` matches Image (or other subclasses).